x-pack/legacy/plugins/infra/server/lib/adapters/framework/kibana_framework_adapter.ts

```diff
@@ -174,28 +175,33 @@ export class InfraKibanaBackendFrameworkAdapter implements InfraBackendFramework
   ) {
     const internalRequest = req[internalInfraFrameworkRequest];
     const server = internalRequest.server;
+    const getVisData = get(server, 'plugins.metrics.getVisData');
+    if (!isFunction(getVisData)) {
+      throw new Error('TSVB is not available');
+    }
 
     // getBasePath returns randomized base path AND spaces path
     const basePath = internalRequest.getBasePath();
     const url = `${basePath}/api/metrics/vis/data`;
 
-    const request = {
-      url,
-      method: 'POST',
-      headers: internalRequest.headers,
-      payload: {
-        timerange,
-        panels: [model],
-        filters,
-      },
-    };
-
-    const res = await server.inject(request);
-    if (res.statusCode !== 200) {
-      throw res;
-    }
-
-    return res.result as InfraTSVBResponse;
+    // For the following request we need a copy of the instnace of the internal request
+    // but modified for our TSVB request. This will ensure all the instance methods
+    // are available along with our overriden values
+    const request = Object.assign(
+      Object.create(Object.getPrototypeOf(internalRequest)),
+      internalRequest,
+      {
+        url,
```

Comment:
Do we still need to override that URL here? The URL doesn't seem to be correct if a `server.basePath` is set but `server.rewriteBasePath` is `false` as described in #45112. It works in any case because `request.url` is not used anywhere in TSVB, but could still be broken by changes to TSVB.

Reply:
Tricky... If we didn't override then the URL would be `{basePath}/api/infra/metrics` and that for sure would break TSVB if they every used the URL internally. Are you thinking if `server.rewriteBasePath` is `false` we should omit the `basePath` in the URL that we are overwriting or do you think we should just leave it as is? We have integration tests that would catch any changes upstream to TSVB if they started using `request.url`.

Reply:
That's how I understand the interaction between these two settings. But on second thought maybe leaving off the base path is not a good idea either since it contains the space identifier. It really depends on how the URL will be used inside of TSVB. Maybe keeping it like this and using the functional tests as a safety net is indeed the best approach.

Reply:
Let's do that... I think this code will change with the new platform anyways, what we are really doing with that request object is forwarding the security credentials. From what I hear that is going to change.